File: MoveHome.java
// represents a move that starts on one of the home rows
class MoveHome implements IMove {
    Pawn pawn;
    int start;
    int distance;
    IMove.MoveType mt = IMove.MoveType.HOME;

    MoveHome(Pawn pawn, int start, int distance) {
        this.pawn=pawn;
        this.start=start;
        this.distance=distance;
    }

    public Pawn get_pawn() {
        return this.pawn;
    }

    public int get_start() {
        return this.start;
    }
    
    public int get_distance() {
        return this.distance;
    }

    public IMove.MoveType get_type() {
        return this.mt;
    }
}

File: python/MoveHome.py
"""
Implements the MoveHome class (expands Move)
"""

#Global

#Local
from Move import Move
from Pawn import Pawn

class MoveHome(Move):
    def __init__(self, pawn, start, distance):
        Move.__init__(self, pawn)
        self.start = start
        self.distance = distance

    def __eq__(self, other):
        if other == None:
            return False
        return self.pawn == other.pawn and self.start == other.start and self.distance == other.distance

if __name__ == "__main__":
    print("The MoveHome class")
